src/tbp/monty/frameworks/models/evidence_unsupervised_inference_matching.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2025 Thousand Brains Project
+#
+# Copyright may exist in Contributors' modifications
+# and/or contributions to the work.
+#
+# Use of this source code is governed by the MIT
+# license that can be found in the LICENSE file or at
+# https://opensource.org/licenses/MIT.
+
+from tbp.monty.frameworks.models.evidence_matching import MontyForEvidenceGraphMatching
+
+
+class MontyForUnsupervisedEvidenceGraphMatching(MontyForEvidenceGraphMatching):
+    """This class removes explicit reset logic in unsupervised inference experiments."""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        # This keeps track of whether the first `pre_episode` is called.
+        # The experiment will not run unless the first `pre_episode` function
+        # is called, which suggests that some class initialization code is
+        # inside `pre_episode`. Initialization and resetting code should be separate.
+        # TODO: Remove initialization logic from `pre_episode`
```

Comment:
I'm not sure I would phrase it this way. `pre_episode` can contain any code that should be executed before an episode. For all our experiments so far, this was both initializing and resetting. For this Monty class we just want to initialize at the beginning of the experiment and do no initialization or resetting in `pre_episode`. It seems a bit messy to put this code into the `pre_episode` call and then do the if statement. Why not put initialization into `pre_epoch` or `pre_eval` or even just into Monty's `init` function for this custom class? Then overwrite `pre_episode` to only update the terminal state and target for logging.

Reply:
> I think these should be defined in the `init` function and reset in `pre_episode`.

I agree, and it's something I've noticed all over the codebase especially since PyCharm by default warns about all instance variables that aren't first defined in an `__init__` method somewhere in the inheritance tree. I've even seen a few places where an object of one class reaches into an object of another class and adds instance variables that didn't exist before (or at least PyCharm thinks so, it's usually pretty accurate). It's a recipe for trouble, but I don't think it's high on our priorities as far as I know.

Reply:
Thanks for the explanation Ramy. Like we discussed in our meeting on Monday, I think in order for this not to become it's whole own project you can just update the comment to include this context for when we look into tackling this TODO some time in the future.